### PR TITLE
Vector Addition 

### PR DESCRIPTION
**Pull request by hrithik-hs**

### CODE REVIEW

**Comment by satu0king** [comment]
Looks good, However if I move the mouse fast, it is not dragging correctly, can you fix that ?

**Comment by satu0king** [comment]
Looks good, However if I move the mouse fast, it is not dragging correctly, can you fix that ? You can probably make it like once the mouse is clicked on the circle, the point will follow the mouse until the mouse is released.

**Comment by hrithik-hs** [comment]
I have tried doing it. However i was unable to do so. Could you suggest some ways to do so.

**Comment by satu0king** [comment]
> I have tried doing it. However i was unable to do so. Could you suggest some ways to do so.

You can see how dragging is handled in CircuitVerse. However in short, you will need to keep track of what thing you clicked on and update it accordingly. I recommend using OOP, so you can do things like having multiple vectors. 

Update: maybe the code here will be more helpful - https://github.com/satu0king/lightxlabHTML5, See function Button on line 691

**Comment by satu0king** [comment]
I would like to see this in a OOP version where new vectors can be dynamically added.

**Comment by hrithik-hs** [comment]
I have fixed the speed issue of updating the vector can you look at it again. I have not yet implemented the OOP version as for now I intend to work with only two vectors. But I'll soon try to add that as well.

**Comment by satu0king** [comment]
> I have fixed the speed issue of updating the vector can you look at it again. I have not yet implemented the OOP version as for now I intend to work with only two vectors. But I'll soon try to add that as well.

I know, I feel it will be a good example of OOP for them.

**Comment by satu0king** [comment]
Add some comments explaining the code, and then it can be merged

File: index.html [hunk 1]
```diff
@@ -8,20 +8,115 @@
...
+        }
         
     }
-
     // Override functions here; 
     canvas.mainFunction = main;
-
-    var timeStep = 50;
+    canvas.setupFunction = setup;
+    var timeStep = 0.5;
```
Comment (satu0king) [inline index.html]: TimeStep cannot be 0.5!

**Comment by satu0king** [comment]
> Add some comments explaining the code, and then it can be merged

Can merge only if code is documented. And author's name should be present, see other people's examples for reference.

**Comment by hrithik-hs** [comment]
I have added the OOP implementation of vector and the comments can you please check now.

**Comment by satu0king** [comment]
I have merged it

**Pull request closed without merging**